### PR TITLE
Add sorting to table component

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue

#286

## Overview
Adds the ability to sort a table column by clicking the header cell.

## Reason
When analysing data it will be necessary to change the order to compare.

## Work carried out
- [x] Add sort
- [x] Update develop documentation
- [x] Update design documentation
- [x] Add sort icons

### Desktop
Screenshot 2019-10-31 at 07 49 35

### Mobile
Screenshot 2019-10-31 at 07 49 47

## Developer notes
- Take a close look at the SVG implementation

### CODE REVIEW

**Comment by hxltrhuxze** [comment]
Really nice :) Having played with the live demo, I think there's possibly a couple of things we should fix to make it more usable.

1. Firstly, it's difficult to know that a column is sortable without clicking on the column header. Perhaps an icon like this could be used?

Screenshot 2019-10-29 at 15 46 14

2. Secondly, is there a way to deselect a table column _after_ it's been sorted? This may be fixed by point 1, as after two clicks on the table header we could revert the column back to its default and swap the icon back to the icon I've linked above.

File: packages/css-framework/src/components/_table.scss [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+  table-layout: fixed;
   width: 100%;
   th {
     font-weight: bold;
     text-align: left;
     @include font-size(xs);
     text-transform: uppercase;
     padding: spacing(4) spacing(2);
+
+    &.sortable {
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_table.scss]: Could we change this stateful class to `is-sortable`? Just for consistency.

File: packages/icon-library/src/assets/table/sort-ascending.svg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<svg width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 9 14">
```
Comment (m7kvqbe1) [inline packages/icon-library/src/assets/table/sort-ascending.svg:1]: Could we not use these ones already in the library?

https://material.io/resources/icons/?search=down&icon=arrow_drop_down&style=round
https://material.io/resources/icons/?search=up&icon=arrow_drop_up&style=round

Maybe they aren't fit for purpose?

File: packages/react-component-library/src/components/Table/Column.tsx [hunk 3]
```diff
@@ -1,10 +1,53 @@
...
+  IconSortDescending,
+  IconSortUnsorted,
+} from '@royalnavy/icon-library'
+
+import { ASCENDING, DESCENDING } from './constants'
 
 export interface ColumnProps {
   field: string
+  sortable?: boolean
+  onSortClick?: Function
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Table/Column.tsx]: Is it worth defining the signature and being more explicit with this type?

**Review by m7kvqbe1 — APPROVED** [review]
Looks good to me - just a couple of small comments

File: packages/icon-library/src/assets/table/sort-ascending.svg (hunk 2 above)
Comment (thyhjwb6) [reply packages/icon-library/src/assets/table/sort-ascending.svg:1]: I originally used the icons from the existing library and through discussion with @hxltrhuxze we are using a different icon for `ascending`, `descending` and `unsorted`.

https://github.com/Royal-Navy/standards-toolkit/pull/368#issuecomment-547488735

**Pull request merged**